### PR TITLE
scRNA

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
Combined effort of @Rebecza and @tilschaef 

**Checklist**
- [x] I made a PR to develop (not master)
- [ ] If applicable: I updated the docs
- [ ] I updated the CHANGELOG
- [ ] These changes are covered by the tests

### CODE REVIEW

File: seq2science/rules/get_genome.smk [hunk 1]
```diff
@@ -3,6 +3,18 @@ import genomepy
...
+config["genomepy_temp"] = ["annotation.gff.gz"]
+
+# add annotation to the expected output if it is required
+if "rna_seq" == get_workflow() or config["aligner"] == "star" or \
+        "scrna_seq" == get_workflow():
+    config["genome_types"].extend(["annotation.gtf", "annotation.bed"])
+
+
+# TODO: return to checkpoint get_genome when checkpoints are stable
+#  1) does the trackhub input update? 2) does ruleorder work?
```
Comment (siebrenf) [inline seq2science/rules/get_genome.smk:17]: can be removed.

File: seq2science/rules/trimming.smk [hunk 2]
```diff
@@ -78,7 +81,20 @@ if config["trimmer"] == "trimgalore":
...
-    ruleorder: fastp_PE > fastp_SE
+    if "scrna_seq" == get_workflow():
+        ruleorder: fastp_SE > fastp_PE
+    else:
+        ruleorder: fastp_PE > fastp_SE
+
+
+    if get_workflow() == "scrna_seq":
+        all_single_samples = [sample for sample in all_samples if sampledict[sample]["layout"] == "SINGLE"]
+        assert len(all_single_samples) == 0
```
Comment (siebrenf) [inline seq2science/rules/trimming.smk:92]: either remove the assert, or add a warning behind it.

**Review by siebrenf — APPROVED** [review]
2 small remarks. looks great otherwise!

**Comment by Maarten-vd-Sande** [comment]
I merged the previous PR and the idea is that @tilschaef will finish this in a PR to this one. So both Rebecca and Tilman get credits for working on it.

It's not finished yet! But the feedback is something @tilschaef can take along when adding 10X support :)

**Comment by Maarten-vd-Sande** [comment]
These same tests seem to be failing on independent PRs... :thinking:

**Comment by Maarten-vd-Sande** [comment]
One thing to change is the samples.tsv for scRNA since the docs/DAG image can not be built for that one. It requires a public sample in the samples.tsv

**Pull request merged**